onmt/tests/test_beam_search.py

```diff
@@ -617,5 +622,6 @@ def test_beam_lm_update_memory_length_when_finished(self):
         self.finish_first_beam_step(beam)
 
         n_steps = beam.alive_seq.shape[-1] - 1
+
         self.assertTrue(beam.memory_lengths.equal(
-            n_steps+fn_map_state(src_lengths[1:], dim=0)))
+            n_steps+fn_map_state(src_lengths, dim=0)))
```

Comment:
The purpose of the test is to check everything is ok when all the beam of an example end. Here it does not happen.
If the method finish_first_beam_step I force all the scores of the beams of eos for the first example. Why is the beam not ending then ?

Reply:
This is not true. Out of the BEAM_SZ=5 beams only 3 out of 5 finish, which means that when the exit condition is ">=n_best" with n_best>=4, according to before this PR, or " >=beam_size" taking into account this PR, then no example is completely finished, and none are removed, hence memory_lengths still contains 3 examples.
This test, as well as some others are not 100% bullet proof wrt the existing code.